core/main/src/firebolt/handlers/voice_guidance_rpc.rs

```diff
@@ -250,35 +253,45 @@ impl VoiceguidanceServer for VoiceguidanceImpl {
         ctx: CallContext,
         request: ListenRequest,
     ) -> RpcResult<ListenerResponse> {
-        // Register for individual change events (no-op if already registered), handlers emit VOICE_GUIDANCE_SETTINGS_CHANGED_EVENT.
-        voice_guidance_settings_enabled_changed(&self.state, &ctx, &request, VOICE_GUIDANCE_CHANGED)
-            .await
+        voice_guidance_settings_enabled_changed(&self.state, &ctx, &request).await
     }
 
     async fn voice_guidance_settings_speed_rpc(&self, _ctx: CallContext) -> RpcResult<f32> {
-        StorageManager::get_number_as_f32(&self.state, StorageProperty::VoiceguidanceSpeed).await
+        voice_guidance_settings_speed(&self.state).await
     }
 
     async fn voice_guidance_settings_speed_set(
         &self,
         _ctx: CallContext,
         set_request: SetF32Property,
     ) -> RpcResult<()> {
-        if set_request.value >= 0.1 && set_request.value <= 10.0 {
-            StorageManager::set_number_as_f32(
-                &self.state,
-                StorageProperty::VoiceguidanceSpeed,
-                set_request.value,
-                None,
-            )
-            .await?;
-
+        if (0.50..=2.0).contains(&set_request.value) {
             let resp = self
                 .state
                 .get_client()
                 .send_extn_request(DeviceInfoRequest::SetVoiceGuidanceSpeed(set_request.value))
                 .await;
             if resp.is_ok() {
+                // TODO: Thunder doesn't currently support a speed change listener so we emit an event here.
```

Comment:
Nitpick: Can we remove the reference to Thunder in the comment. Ideally we want the setter to be implemented in the SettingsProcessor as part of a new Set enum in `SettingsRequest` but it might be just additional work for this effort.